units/host_allocator.go

```diff
@@ -254,33 +255,11 @@ func (j *hostAllocatorJob) Run(ctx context.Context) {
 	const lowRatioThresh = float32(.25)
 	terminateExcess := distro.HostAllocatorSettings.HostsOverallocatedRule == evergreen.HostsOverallocatedTerminate
 	if terminateExcess && hostQueueRatio < lowRatioThresh && len(upHosts) > 0 {
-
-		var killableHosts, newCapTarget int
-		if hostQueueRatio == 0 {
-			killableHosts = len(upHosts)
-			newCapTarget = 0
+		distroIsByHour := cloud.UsesHourlyBilling(&upHosts[0].Distro)
+		if distroIsByHour {
```

Comment:
I guess the code didn't quite organize in the way I was envisioning. If you don't need to pass `upHosts` to `setTargetAndTerminate`, you should probably just use `if !distroIsByHour { j.setTargetAndTerminate(...) }` rather than return, since the job might do something else after this block.